config/src/lib.rs

```diff
@@ -1,14 +1,57 @@
-pub fn add(left: usize, right: usize) -> usize {
-    left + right
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize, PartialEq)]
+enum Performances {
+    High,
+    Average,
+    Low,
+}
+
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct Config {
+    performance: Performances,
+    width: Option<u16>,
```

Comment:
Believe is better have as u16 instead of Option<u16>. To avoid additional unwrap when using the configuration struct later. Since all properties should have a default we can unwrap it on the new.

Also will be good to have a default in the impl. E.g: https://doc.rust-lang.org/std/default/trait.Default.html

Reply:
I have started the implementation [here](https://github.com/raphamorim/rio/pull/7/commits/461d008b8162707150771e00dc966615c77f9684)

I couldn't come with a solution for when the user have the config file, but didn't pass one of the parameters.

Because if I set the struct as below:
```rust
struct Config {
    performance: Option<Performance>,
    width: u16,
    height: u16,
}
```

Now the problem is that when we parse, as the `user` did not specified performance, it is `None`, did you have something in mind?

the code can be not that rusty, still learning some patterns, if you think something can improve, just let me know

Reply:
No worries at all @1garo, you are doing a great job. Usually when you're wrapping and come as None, you can use `unwrap_or_else(||` to return the default. Let me know if you still have questions, we can book a day and do this part together 👍

Reply:
sounds great, if you can talk today or tomorrow, we can do it

Reply:
If you have telegram, just drop me a message on same username as GH. I also have discord just in case.

Reply:
I think discord is easier for me, I used telegram once in my life. 1garo#7086